File: TDDMicroExercises.YoursSolutions/Sam-Serpoosh/code/turn_ticket_dispenser/ticket_dispenser.rb
require_relative "./turn_number_sequence"
require_relative "./turn_ticket"

class TicketDispenser

  def initialize(ticket_sequence_generator=TurnNumberSequence)
    @ticket_sequence_generator = ticket_sequence_generator
  end

  def get_turn_ticket 
    new_turn_number = @ticket_sequence_generator.get_next_turn_number
    TurnTicket.new(new_turn_number)
  end
end

File: TDDMicroExercises.YoursSolutions/Java-viswans/TurnTicketDispenser/src/main/java/tddmicroexercises/turnticketdispenser/TicketDispenser.java
package tddmicroexercises.turnticketdispenser;

public class TicketDispenser
{
	
	private ITurnNumberSequence seq;
	
	@Deprecated
	public TicketDispenser() {
		this.seq = new ITurnNumberSequence() {
			
			public int getNextTurnNumber() {
				return TurnNumberSequence.getNextTurnNumber();
			}
			
		};
	}
	
	public TicketDispenser(ITurnNumberSequence seq) {
		this.seq = seq;
	}
	
    public TurnTicket getTurnTicket()
    {
        return _getTurnTicket();
    }
    
    private TurnTicket _getTurnTicket() {
    	int newTurnNumber = seq.getNextTurnNumber();
        TurnTicket newTurnTicket = new TurnTicket(newTurnNumber);
        return newTurnTicket;
    }
    
}
